File: EPI/7.6/swift/7.6_test.swift
import Darwin

let fns = [reverseWordsSimple, reverseWordsSmart]

let tests: [(before: String, after: String)] = [
    ("", ""),
    ("a", "a"),
    ("ab", "ab"),
    ("ab c", "c ab"),
    ("bla blah blahhhh", "blahhhh blah bla"),
]

for fn in fns {
    for test in tests {
        var actualCharacters = [Character](test.before)
        fn(&actualCharacters)
        let actualString = String(actualCharacters)
        guard actualString == test.after else {
            print("For string '\(test.before)', expected final string to be '\(test.after)', " +
                "but was '\(actualString)'")
            exit(1)
        }
    }
}

File: EPI/7.6/typescript/7.6_test.ts
import reverseWords from './7.6';

declare function require(name: string): any;
const assert = require('assert');

const tests: {input: string, output: string}[] = [
    {input: "", output: ""},
    {input: "a", output: "a"},
    {input: "ab", output: "ab"},
    {input: "ab c", output: "c ab"},
    {input: "bla blah blahhhh", output: "blahhhh blah bla"},
];

for (const test of tests) {
    const actual = reverseWords(test.input);
    assert.strictEqual(actual, test.output, `For input ${test.input}, ` +
        `expected output to be ${test.output}, but was ${actual}`);
}
